sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -357,6 +346,21 @@ class SessionCatalog(
     tempTables.remove(formatTableName(name))
   }
 
+  /**
+   * Retrieve the metadata of an existing temporary view.
+   * If the temporary view does not exist, return None.
+   */
+  def getTempViewMetadataOption(name: String): Option[CatalogTable] = synchronized {
```

Comment:
Seems we always use it with the pattern `getTempViewMetadataOption.getOrElse(getTableMetadata)`, maybe we should just rename it to `getTempViewOrPermanentTableMetadata`?